packages/gateway-edge-nextjs/src/cal/cal.ts

```diff
@@ -0,0 +1,115 @@
+import { type NextApiRequest, type NextApiResponse } from "next"
+import ical, { type ICalEventData } from "ical-generator";
+import { createServerSideHelpers } from "@trpc/react-query/server"
+import { appRouter, createContextInner, transformer } from "@dotkomonline/gateway-trpc"
+
+const helpers = createServerSideHelpers({
+  router: appRouter,
+  ctx: await createContextInner({
+    auth: null,
+  }),
+  transformer, // optional - adds superjson serialization
+})
+
+function eventUrl(event: { id: string }) {
+  // a better to to get/configure the url?
+  return `https://new.online.ntnu.no/events/${event.id}`;
+}
+
+// ALL events
+export async function CalendarAll(req: NextApiRequest, res: NextApiResponse) {
+  if (req.method !== "GET") {
+    res.status(405).end()
+    return
+  }
+
+  const events = await helpers.event.all.fetch()
+  const instance = ical({ name: `Online Linjeforening Arrangementer` });
+
+  events.forEach(event => {
+    instance.createEvent(toICal(event));
+  });
+
+
+  res.status(200).send(instance.toString())
+}
+
+// a single event
+export async function CalendarEvent(req: NextApiRequest, res: NextApiResponse) {
+  if (req.method !== "GET") {
+    res.status(405).end()
+    return
+  }
+
+  const eventid = req.query.eventid as string;
+  if (!eventid) {
+    res.status(400).json({ message: "Missing eventid" })
+    return
+  }
+
+  const event = (await helpers.event.get.fetch(eventid)).event;
+
+  const instance = ical();
+  instance.createEvent(toICal(event));
+
+  res.status(200).send(instance.toString())
+}
+
+
+// all events a user is attending
+export async function CalendarUser(req: NextApiRequest, res: NextApiResponse) {
+  if (req.method !== "GET") {
+    res.status(405).end()
+    return
+  }
+
+  const userid = req.query.user as string;
+  if (!userid) {
+    res.status(400).json({ message: "Missing user" })
+    return
+  }
+
+
+  // TODO figure out some auth here?
+  // const user = await helpers.user.all.fetch()
+  // console.log(user)
+
+  const events = await helpers.event.allByUserId.fetch({ id: userid })
+  const instance = ical({ name: `${userid} online kalender` });
+
+  events.forEach(event => {
+    instance.createEvent(toICal(event));
+    instance.createEvent(toICal(toRegistration(event)));
+  });
+
+  res.status(200).send(instance.toString())
+}
+
+function toICal(event: { start: Date, end: Date, title: string, description: string | null, location: string | null, id: string }): ICalEventData {
```

Comment:
You can refactor this type to `Pick<Event, ...>` to keep a single source of truth for the Event attributes

Reply:
Ok i did it, pls review